common/services/certificate.js

```diff
@@ -96,27 +101,32 @@ export default {
     getCertificate(ephemeralKey) {
         return new Promise(async function (resolve, reject) {
             const cert = await retrieveSavedCert()
-            if (cert == null && cert == undefined) {
+
+            if (cert) {
+                resolve({state: 'success', message: 'Doordeck SDK is loaded'})
+            } else {
                 registerCertificate(ephemeralKey.publicKey)
                     .then(response => {
                         storeCert(response.data)
                         resolve({state: 'success', message: 'Doordeck SDK is loaded'})
                     })
                     .catch(fail => {
-                        if (fail.response.status === 423) {
-                            if (localStorage.token !== null && localStorage.token !== undefined) {
-                                registerCertificate2FA(ephemeralKey.publicKey).then(response => {
+                        if (fail.response.status === 423 && localStorage.token) {
+                            registerCertificate2FA(ephemeralKey.publicKey)
+                                .then(response => {
                                     if (response.data.method !== undefined && response.data.method !== null) {
                                         reject({state: 'verify', method: response.data.method})
-                                    } else reject({state: 'error', message: 'Failed to send 2FA request'})
+                                    } else {
+                                        reject({state: 'verify', message: 'Failed to send 2FA request'})
+                                    }
+                                })
+                                .catch(fail => {
+                                    reject({state: 'verify', message: 'Failed to send 2FA request'})
                                 })
-                            }
                         } else {
-                            reject({state: 'error', message: 'Failed to get certificate'})
+                            reject({state: 'verify', message: 'Failed to get certificate'})
```

Comment:
Why this?

Reply:
Because we are still at `VerifyRequired` step, basically, `getCertificate` returns two different states it is either `success` or `verify`.
`success` means it is done - completed
`verify` means it needs to do verify again

We don't need an extra state like `error` which caused the initial error on the `app` side